frontend/src/lib/Hooks.ts

```diff
@@ -28,16 +28,23 @@ export const usePrevious = (value: any): any => {
 }
 
 export const useIsOverflowing = (
-  ref: MutableRefObject<HTMLElement | null>
+  ref: MutableRefObject<HTMLElement | null>,
+  checkHeight = true
 ): boolean => {
   const { current } = ref
   const [isOverflowing, setIsOverflowing] = useState(false)
 
   useEffect(() => {
     if (current) {
-      const { scrollHeight, clientHeight } = current
-
-      setIsOverflowing(scrollHeight > clientHeight)
+      if (checkHeight) {
+        // Check height overflow
+        const { scrollHeight, clientHeight } = current
+        setIsOverflowing(scrollHeight > clientHeight)
+      } else {
+        // Check width overflow
+        const { scrollWidth, clientWidth } = current
+        setIsOverflowing(scrollWidth > clientWidth)
+      }
     }
   })
```

Comment:
no strong opinion on this, but I wonder if the API design would be cleaner if this hook instead returned and object of type `{ vertical: boolean, horizontal: boolean }` with the overflow states of both rather than taking an argument

Reply:
Sounds good 👍  I refactored it to return both values

Reply:
I decided to roll back all my changes for the `useIsOverflowing` since I also need to make the effect depending on the count of items (otherwise this causes CPU issues for the tabs). Maybe this is better for a dedicated refactoring.